### ISSUE TITLE
Support tagged template

### ISSUE THREAD

**Issue by jraoult** [unplugin/unplugin-macros#27]
### Clear and concise description of the problem

I tried to emulate a static (aka compile stime) CSS bundler using `unplugin-macros`. Some CSS in JS solutions rely on tagged templates:
```js
className={css`
  color: red;
`}
```

For example I know [Panda css](https://panda-css.com/docs/concepts/template-literals), [Emotion](https://emotion.sh/docs/@emotion/css#string-styles) and others do. 

I tried to export a macro like that:

```js
export const css = (templateStrings)=> {
   [...]
}
```

and then use it:
```js
import { css } from "./macro" with { type: "macro" };

console.log(css`
  color: red;
`);
```

But I get the following error with Vite (tested in `v5.0.8` and `v5.1.4`), making it impossible to use a tagged template:

```bash
[unplugin-macros] Cannot read properties of undefined (reading 'start')
```

Is there something I'm missing and an easy way to make it work? If not, would you consider supporting it?

### Suggested solution

I'm not an AST expert but I suspect it would require adding a branch to check if the exported macro is called as a tagged template and execute it in the file https://github.com/unplugin/unplugin-macros/blob/main/src/core/index.ts

### Validations

- [X] Follow our [Code of Conduct](https://github.com/unplugin/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by sxzz** [comment]
Released as `v0.9.3`

**Comment by jraoult** [comment]
Wow! Wasn't expecting such a quick turnaround. Let me try that.
